### PR TITLE
[Docs] Update docker container doc

### PR DESCRIPTION
**Pull request by cblmemo**
This PR updates the docker container doc by:
* give a concrete example of SKYPILOT_DOCKER_SERVER;
* suggest passing secrets

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/examples/docker-containers.rst [hunk 1]
```diff
@@ -27,27 +27,25 @@ For example, to use the :code:`ubuntu:20.04` image from Docker Hub:
...
+We suggest to setting the :code:`SKYPILOT_DOCKER_PASSWORD` environment variable through the CLI (see :ref:`passing secrets <passing-secrets>`):
 
 .. code-block:: bash
 
-  $ sky launch \
-      --env SKYPILOT_DOCKER_PASSWORD=$(aws ecr get-login-password --region us-east-1) \
-      --env SKYPILOT_DOCKER_USERNAME=AWS \
-      private_docker.yaml
+  $ export SKYPILOT_DOCKER_PASSWORD=$(aws ecr get-login-password --region us-east-1)
+  $ sky launch ecr_private_docker.yaml --env SKYPILOT_DOCKER_PASSWORD
```
Comment (romilbhardwaj) [inline docs/source/examples/docker-containers.rst:48]:
```suggestion
  $ sky launch ecr_private_docker.yaml --env $SKYPILOT_DOCKER_PASSWORD
```

File: docs/source/examples/docker-containers.rst [hunk 2]
```diff
@@ -27,27 +27,25 @@ For example, to use the :code:`ubuntu:20.04` image from Docker Hub:
...
 
-For Docker images hosted on private registries, you can provide the registry authentication details using the :code:`envs` section:
+For Docker images hosted on private registries, you can provide the registry authentication details using :ref:`task environment variables <env-vars>`:
 
 .. code-block:: yaml
 
-  # private_docker.yaml
+  # ecr_private_docker.yaml
   resources:
     image_id: docker:<your-private-image>:<tag>
```
Comment (romilbhardwaj) [inline docs/source/examples/docker-containers.rst:36]: If you're giving a concrete example, might as well also update this to something concrete: 
```suggestion
    image_id: docker:<your-user-id>.dkr.ecr.us-east-1.amazonaws.com/your-private-image:latest
```

**Review by romilbhardwaj — APPROVED** [review]

File: docs/source/examples/docker-containers.rst (hunk 2 above)
Comment (cblmemo) [reply docs/source/examples/docker-containers.rst:36]: Oh actually we only require users to specify `your-private-image` in the `image_id` section... See [here](https://github.com/skypilot-org/skypilot/blob/e5e400ba9b2b140866cbb8c9e3e4bb88df18dd33/sky/skylet/providers/command_runner.py#L146)

Comment (cblmemo) [reply docs/source/examples/docker-containers.rst:36]: I see the confusion. Maybe we could support both? Let me raise an PR to fix it

Comment (cblmemo) [reply docs/source/examples/docker-containers.rst:36]: Done in #2574

**Pull request merged**